snabbdom.js

```diff
@@ -30,14 +31,18 @@ function createKeyToOldIdx(children, beginIdx, endIdx) {
 
 function createRmCb(childElm, listeners) {
   return function() {
-    if (--listeners === 0) childElm.parentElement.removeChild(childElm);
+    if (--listeners === 0) {
+      var parent = api.parentNode(childElm);
+      api.removeChild(parent, childElm);
+    }
   };
 }
 
 var hooks = ['create', 'update', 'remove', 'destroy', 'pre', 'post'];
 
-function init(modules) {
+function init(modules, domApi) {
```

Comment:
This is nitpicking but I think the variable names are slightly confusing. We call the actual DOM api `api` and the user supplied API `domApi`. But the user supplied interface implementation might have nothing to do with the DOM while the DOM interface certainly does.

What about renaming `api` to `domApi` and `domApi` to `userApi`. Alternatively rename `api` to `domImpl` and `domApi` to `apiImpl`. The former is probably best though.